packages/btc/src/api/sendRbf.ts

```diff
@@ -0,0 +1,169 @@
+      return {
+        script: output.script,
+        value: output.value,
+        fixed: true,
+      };
+    } else {
+      return {
+        minUtxoSatoshi: requireValidOutputsValue ? undefined : output.value,
+        address: bitcoin.address.fromOutputScript(output.script, network),
+        value: output.value,
+        fixed: true,
+      };
+    }
+  });
+
+  // Set change output if specified
+  let changeAddress: string | undefined = props.changeAddress;
+  if (props.changeIndex !== undefined) {
+    const changeOutput = outputs[props.changeIndex];
+    if (!changeOutput) {
+      throw TxBuildError.withComment(ErrorCodes.INVALID_CHANGE_OUTPUT, `outputs[${props.changeIndex}] is not found`);
+    }
+    const isReturnableOutput = changeOutput && 'address' in changeOutput;
+    if (!isReturnableOutput) {
+      throw TxBuildError.withComment(
+        ErrorCodes.INVALID_CHANGE_OUTPUT,
+        `outputs[${props.changeIndex}] is not a returnable output for change`,
+      );
+    }
+    const changeOutputAddress = isReturnableOutput ? changeOutput.address : undefined;
+    if (changeOutputAddress && changeAddress && changeAddress !== changeOutputAddress) {
+      throw TxBuildError.withComment(
+        ErrorCodes.INVALID_CHANGE_OUTPUT,
+        `outputs[${props.changeIndex}].address does not match the specified changeAddress`,
+      );
+    }
+    if (changeOutputAddress && !changeAddress) {
+      changeAddress = changeOutputAddress;
+    }
+
+    const config = networkTypeToConfig(props.source.networkType);
+    const minUtxoSatoshi = props.minUtxoSatoshi ?? config.btcUtxoDustLimit;
+    changeOutput.minUtxoSatoshi = minUtxoSatoshi;
+    changeOutput.value = minUtxoSatoshi;
+    changeOutput.protected = true;
+    changeOutput.fixed = false;
+  }
+
+  // Fee rate
+  const requireGreaterFeeAndRate = props.requireGreaterFeeAndRate ?? true;
+  let feeRate: number | undefined = props.feeRate;
+  if (requireGreaterFeeAndRate && !feeRate) {
+    const feeRates = await props.source.service.getBtcRecommendedFeeRates();
+    feeRate = feeRates.fastestFee;
+  }
+
+  // The RBF transaction should offer a higher fee rate
+  const previousInsValue = inputs.reduce((sum, input) => sum + input.value, 0);
+  const previousOutsValue = previousTx.outs.reduce((sum, output) => sum + output.value, 0);
+  const previousFee = previousInsValue - previousOutsValue;
+  const previousFeeRate = Math.floor(previousFee / previousTx.virtualSize());
+  if (requireGreaterFeeAndRate && feeRate !== undefined && feeRate <= previousFeeRate) {
+    throw TxBuildError.withComment(
+      ErrorCodes.INVALID_FEE_RATE,
+      `RBF should offer a higher fee rate, previous: ${previousFeeRate}, current: ${feeRate}`,
+    );
+  }
+
+  // Exclude the previous transaction's outputs during the collection
+  const previousTxId = previousTx.getId();
+  const excludeUtxos: BaseOutput[] = previousTx.outs.map((_, index) => ({
+    txid: previousTxId,
+    vout: index,
+  }));
```

Comment:
TODO: If `props.onlyConfirmedUtxos === true`, then all the previousTxId's child txs's UTXO should also be excluded.

It may be a rare case.